### DISCUSSION TITLE
delete get cache best practices?

### DISCUSSION THREAD

**Discussion by andrew2558** [arthurfiorette/axios-cache-interceptor#742] [Q&A] [answered]
When I initiate a POST operation, I want to clear the cache for a specific GET request, but I don't know the id associated with that GET request. Is there a way to know the id generated by axios-cache-interceptor? 
Additionally, is there a method to clear the cache for a specific GET request without using an id? Setting an id for each GET request seems not to be a good practice, especially when I don't know the system-generated id. Are there any best practices for clearing the cache after an update?

**Comment by arthurfiorette** [comment]
You need to know the request ID, otherwise there's nothing you can do to delete the correct request. A good pattern I'm currently using is to manually define all requests IDS inside an enum and use them accordingly where I need.


```ts
const IDS = {
  createUser: 'create-user',
  listUsers: 'list-users'
}

function listUsers() {
  return axios.get('url...', { id: IDS.listUsers, <other options> });
}

function createUser(data) {
  return axios.post('url', data, {
    id: IDS.createUser, 
    update: {
      [IDS.listUsers]: 'delete'
    }
  });
}
```

**Reply by andrew2558** [reply]
thanks, @arthurfiorette 
I have a question: If loading paginated data with the same API endpoint but different parameters, will the system correctly handle situations where different page data loads using the same ID?

**Reply by arthurfiorette** [reply, marked as answer]
You could've tested it out... But yes, you have to provide different ID for each parameter, you can do something like:

```ts
const IDS = {
  getUserById: (id) => `get-user-by-id-${id}`,
  listUsers: 'list-users'
}

function getUserById(userId) {
  return axios.get(`url/${userId}`, { id: IDS.getUserById(userId), <other options> });
}
```

**Reply by arthurfiorette** [reply]
@andrew2558 if this solves your issue, could you mark it as answer?

**Reply by andrew2558** [reply]
@arthurfiorette 
thank you very much.
